conversant/prompt_chatbot.py

```diff
@@ -187,27 +188,39 @@ def _update_max_context_examples(
             "tokens"
         )
 
+    def _dispatch_concurrent_generate_call(self, **kwargs) -> Future:
+        """Dispatches a concurrent call to co.generate.
+
+        This allows a network bound co.generate call to proceed while also
+        yielding the current response in a partial reply generator.
+
+        Args:
+            kwargs: Keyword arguments for the call to co.generate.
+
+        Returns:
+            Future: A future object that will be called to retrieve the result of
+                co.generate.
+        """
+        with ThreadPoolExecutor(max_workers=1) as exe:
+            future = exe.submit(self.co.generate, **kwargs)
+        return future
+
     def get_stop_seq(self, response: str) -> str:
         """Given a response, returns the stop sequence it ends with if any.
 
         Args:
             response (str): Response coming from prompt chatbot.
 
         Returns:
-            stop_seq (str): The stop sequence at the end of response.
+            str: The stop sequence at the end of response. If no stop
+                sequence is found, then an empty string is returned.
 
         """
         for stop_seq in self.client_config["stop_sequences"]:
-            if response.endswith(stop_seq):
+            if stop_seq in response:
                 return stop_seq
-
-        # If there is not a stop sequence at the end
         return ""
 
-    def partial_reply_max_reruns(self) -> int:
```

Comment:
Since this function is only called once in `partial_reply`, it need not be factored out to its own function (and it is a straightforward function too).